server/plugin/user_test.go

```diff
@@ -0,0 +1,399 @@
+
+			p.store = mockedStore
+
+			user, err := p.GetUser("mock-userID")
+
+			if testCase.expectedErr != "" {
+				require.Nil(t, user)
+				require.EqualError(t, err, testCase.expectedErr)
+			} else {
+				require.Nil(t, err)
+				require.IsTypef(t, &serializer.User{}, user, "mockMsg")
+			}
+		})
+	}
+}
+
+func Test_DisconnectUser(t *testing.T) {
+	for _, testCase := range []struct {
+		description string
+		errMessage  error
+		expectedErr string
+	}{
+		{
+			description: "User is deleted successfully from KV store using mattermostID",
+			errMessage:  nil,
+		},
+		{
+			description: "Error in deleting user from KV store using mattermostID",
+			errMessage:  errors.New("mockErrMessage"),
+			expectedErr: "mockErrMessage",
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			p := Plugin{}
+
+			mockCtrl := gomock.NewController(t)
+			mockedStore := mock_plugin.NewMockStore(mockCtrl)
+
+			mockedStore.EXPECT().DeleteUser("mock-userID").Return(testCase.errMessage)
+
+			p.store = mockedStore
+
+			err := p.DisconnectUser("mock-userID")
+
+			if testCase.expectedErr != "" {
+				require.EqualError(t, err, testCase.expectedErr)
+			} else {
+				require.Nil(t, err)
+			}
+		})
+	}
+}
+
+func Test_CreateDisconnectUserAttachment(t *testing.T) {
+	t.Run("CreateDisconnectUserAttachment created successfully", func(t *testing.T) {
+		p := Plugin{}
+
+		disconnectUserPath := fmt.Sprintf("%s%s", p.GetPluginURLPath(), PathUserDisconnect)
+		expectedResponse := &model.SlackAttachment{
+			Title: DisconnectUserConfirmationMessge,
+			Color: "#FF0000",
+			Actions: []*model.PostAction{
+				{
+					Type: "button",
+					Name: "Yes",
+					Integration: &model.PostActionIntegration{
+						URL: disconnectUserPath,
+						Context: map[string]interface{}{
+							DisconnectUserContextName: true,
+						},
+					},
+				},
+				{
+					Type: "button",
+					Name: "No",
+					Integration: &model.PostActionIntegration{
+						URL: disconnectUserPath,
+						Context: map[string]interface{}{
+							DisconnectUserContextName: false,
+						},
+					},
+				},
+			},
+		}
+
+		res := p.CreateDisconnectUserAttachment()
+
+		require.EqualValues(t, res, expectedResponse)
+	})
+}
+
+func Test_GetDisconnectUserPost(t *testing.T) {
+	for _, testCase := range []struct {
+		description                      string
+		errMessage                       *model.AppError
+		expectedErr                      string
+		userID                           string
+		getPostWithSlackAttachmentResult *model.Channel
+	}{
+		{
+			description:                      "GetDisconnectUserPost returns proper response",
+			errMessage:                       nil,
+			userID:                           "mock-userID",
+			getPostWithSlackAttachmentResult: &model.Channel{},
+			expectedErr:                      "",
+		},
+		{
+			description:                      "GetDisconnectUserPost return error because GetDirectChannel return error dur to invalid userID",
```

Comment:
```suggestion
			description:                      "GetDisconnectUserPost return error because GetDirectChannel return error due to invalid userID",
```